prog/setup_hugepages.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class Prog::SetupHugepages < Prog::Base
+  subject_is :sshable, :vm_host
+
+  def start
+    hugepage_size = "1G"
+    # TODO: how should we calculate required hugepages?
+    # put away min(4G,mem*0.25) for host, try to reserve rest as hugepages
+    hugepage_cnt = [vm_host.total_mem_gib - 4, vm_host.total_mem_gib * 3 / 4].max
+    sshable.cmd("sudo sed -i '/^GRUB_CMDLINE_LINUX=\"/ s/\"$/ default_hugepagesz=#{hugepage_size} hugepagesz=#{hugepage_size} hugepages=#{hugepage_cnt}&/' /etc/default/grub")
+    sshable.cmd("sudo update-grub")
+    begin
+      sshable.cmd("sudo reboot")
+    rescue
+    end
+    hop :wait_reboot
+  end
+
+  def wait_reboot
+    # TODO: don't wait indefinitely
```

Comment:
Did you have something in mind that we could do other than keep waiting?

Reply:
I think we can let it stick there indefinitely and page probably. Or we come up with bespoke resolution in situ. But today, we can let it be simple, and indefinite.